worlds/sc2/Client.py

```diff
@@ -1364,8 +1364,8 @@ def mission_reqs_completed(ctx: SC2Context, mission_name: str, missions_complete
                 else:
                     req_success = False
 
-            # Grid-specific logic (to avoid long path checks and infinite recursion)
-            if ctx.mission_order in (MissionOrder.option_grid, MissionOrder.option_mini_grid, MissionOrder.option_medium_grid):
+            # Grid and Blitz-specific logic (to avoid long path checks and infinite recursion) (Includes legacy grids)
+            if ctx.mission_order in (MissionOrder.option_grid, MissionOrder.option_blitz, 3, 4, 8):
```

Comment:
Should be tied to slot_data version (the 3, 4, 8 part). Do a version bump there

Reply:
It should be 3, 4, 8 regardless of whether it was generated with the past release or not, though?

Reply:
The check is easy, just use `ctx.slot_data_version`. I'd prefer safe way instead and when checking slot data version, it makes clear, that some code exists for backwards compatibility

```python
if ctx.mission_order in (MissionOrder.option_grid, MissionOrder.option_blitz) or (ctx.slot_data_version <= 3 and ctx.mission_order in (3, 4, 8)):
```
I think this wouldn't introduce any new trap